pkg/controller/populators/forklift-populator.go

```diff
@@ -0,0 +1,644 @@
+		return &v1beta1.OvirtVolumePopulator{}, nil
+	case "OpenstackVolumePopulator":
+		return &v1beta1.OpenstackVolumePopulator{}, nil
+	default:
+		return nil, fmt.Errorf("unknown populator type %T", pvc.Spec.DataSourceRef.Kind)
+	}
+}
+
+func isPVCForkliftKind(pvc *corev1.PersistentVolumeClaim) bool {
+	dataSourceRef := pvc.Spec.DataSourceRef
+	if dataSourceRef == nil {
+		return false
+	}
+
+	if (dataSourceRef.APIGroup != nil && *dataSourceRef.APIGroup != apiGroup) ||
+		dataSourceRef.Name == "" || dataSourceRef.Kind == "" {
+		return false
+	}
+
+	_, ok := supportedPopulators[dataSourceRef.Kind]
+	return ok
+}
+
+func isPVCPrimeForkliftKind(pvc *corev1.PersistentVolumeClaim) bool {
+	owner := metav1.GetControllerOf(pvc)
+	if owner == nil || owner.Kind != "PersistentVolumeClaim" {
+		return false
+	}
+
+	populatorKind := pvc.Annotations[cc.AnnPopulatorKind]
+	return populatorKind == "forklift"
+}
+
+func (r *ForkliftPopulatorReconciler) reconcileTargetPVC(pvc, pvcPrime *corev1.PersistentVolumeClaim) (reconcile.Result, error) {
+	pvcPrimeCopy := pvcPrime.DeepCopy()
+
+	// Look for the populator pod
+	podName := fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID)
+	pod, err := r.getImportPod(pvcPrime, podName)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if pod == nil {
+		err = r.createPopulatorPod(pvcPrime, pvc)
+		if err != nil {
+			if errors.Is(err, errCrNotFound) || k8serrors.IsAlreadyExists(err) {
+				return reconcile.Result{}, nil
+			}
+
+			return reconcile.Result{}, err
+		}
+
+		return reconcile.Result{}, nil
+	}
+
+	if pod.Status.ContainerStatuses == nil || len(pod.Status.ContainerStatuses) == 0 {
+		return reconcile.Result{RequeueAfter: 2 * time.Second}, nil
+	}
+
+	anno := pvcPrimeCopy.Annotations
+	anno[cc.AnnPodPhase] = string(pod.Status.Phase)
+	anno[cc.AnnImportPod] = pod.Name
+	anno[cc.AnnPodRestarts] = strconv.Itoa(int(pod.Status.ContainerStatuses[0].RestartCount))
+
+	phase := pvcPrimeCopy.Annotations[cc.AnnPodPhase]
+	switch phase {
+	case string(corev1.PodRunning):
+		if err := r.updatePVCPrime(pvc, pvcPrimeCopy); err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: 2 * time.Second}, nil
+	case string(corev1.PodFailed):
+		r.recorder.Eventf(pvc, corev1.EventTypeWarning, importFailed, messageImportFailed, pvc.Name)
+	case string(corev1.PodPending):
+		return reconcile.Result{RequeueAfter: 2 * time.Second}, nil
+	case string(corev1.PodSucceeded):
+		if cc.IsPVCComplete(pvcPrime) && cc.IsUnbound(pvc) {
+			// TODO(benny) use a different const?
+			r.recorder.Eventf(pvc, corev1.EventTypeNormal, importSucceeded, messageImportSucceeded, pvc.Name)
+		}
+
+		if err := cc.Rebind(context.TODO(), r.client, pvcPrime, pvc); err != nil {
+			return reconcile.Result{}, err
+		}
+	default:
+		//Should never happen
+		return reconcile.Result{}, fmt.Errorf("unknown pod phase %s", phase)
+	}
+
+	if err := r.updatePVCPrime(pvc, pvcPrimeCopy); err != nil {
+		return reconcile.Result{}, nil
```

Comment:
why change to `return reconcile.Result{}, nil` instead of returning the err?

Reply:
forgot to revert back after experimentation